src/gtc/common.py

```diff
@@ -572,6 +572,46 @@ def apply(cls, node: Node, *, symtable: Dict[str, Any]) -> List[str]:
     return root_validator(allow_reuse=True, skip_on_failure=True)(_impl)
 
 
+def validate_lvalue_dims() -> RootValidatorType:
+    """Validate lvalue dimensions using the root node symbol table."""
+
+    class _LvalueDimsValidator(NodeVisitor):
+        def visit_Node(self, node: Node, symtable: Dict[str, Any], **kwargs: Any) -> None:
+            if hasattr(node, "loop_order"):
+                kwargs["loop_order"] = node.loop_order
+            if isinstance(node, SymbolTableTrait):
+                symtable = {**symtable, **node.symtable_}
+            self.generic_visit(node, symtable=symtable, **kwargs)
+
+        def visit_AssignStmt(
+            self, node: AssignStmt, *, loop_order: LoopOrder, **kwargs: Any
+        ) -> None:
+            symtable = kwargs["symtable"]
```

Comment:
```suggestion
            self, node: AssignStmt, *, loop_order: LoopOrder, *, symtable, **kwargs: Any
        ) -> None:
```
This is the usual pattern elsewhere.

Reply:
👍